static/app/views/dashboardsV2/detail.spec.jsx

```diff
@@ -396,43 +376,29 @@ describe('Dashboards > Detail', function () {
         method: 'PUT',
         body: TestStubs.Dashboard([widgets[0]], {id: '1', title: 'Custom Errors'}),
       });
-      await act(async () => {
-        wrapper = mountWithTheme(
-          <OrganizationContext.Provider value={initialData.organization}>
-            <ViewEditDashboard
-              organization={initialData.organization}
-              params={{orgId: 'org-slug', dashboardId: '1'}}
-              router={initialData.router}
-              location={initialData.router.location}
-            />
-          </OrganizationContext.Provider>,
-          initialData.routerContext
-        );
-        await tick();
-        wrapper.update();
-      });
+      render(
+        <OrganizationContext.Provider value={initialData.organization}>
+          <ViewEditDashboard
+            organization={initialData.organization}
+            params={{orgId: 'org-slug', dashboardId: '1'}}
+            router={initialData.router}
+            location={initialData.router.location}
+          />
+        </OrganizationContext.Provider>,
+        {context: initialData.routerContext}
+      );
 
-      expect(mockVisit).toHaveBeenCalledTimes(1);
+      await waitFor(() => expect(mockVisit).toHaveBeenCalledTimes(1));
 
       // Enter edit mode.
-      wrapper.find('Controls Button[data-test-id="dashboard-edit"]').simulate('click');
+      userEvent.click(screen.getByText('Edit Dashboard'));
```

Comment:
It is a button, are we preferring `getByRole`? I thought it was a slower selector that we were avoiding if we could

Reply:
I think we’ve mostly been using getByRole without a problem so far

I would prefer it I think since it’s much less brittle and does some good accessibility testing